compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -1285,14 +1310,25 @@ struct TupleMemberDescriptionFactory<'tcx> {
 
 impl<'tcx> TupleMemberDescriptionFactory<'tcx> {
     fn create_member_descriptions(&self, cx: &CodegenCx<'ll, 'tcx>) -> Vec<MemberDescription<'ll>> {
+        let capture_names = match *self.ty.kind() {
+            ty::Generator(def_id, ..) | ty::Closure(def_id, ..) => {
+                Some(closure_saved_names_of_captured_variables(cx.tcx, def_id))
+            }
+            _ => None,
+        };
         let layout = cx.layout_of(self.ty);
         self.component_types
             .iter()
             .enumerate()
             .map(|(i, &component_type)| {
                 let (size, align) = cx.size_and_align_of(component_type);
+                let name = if let Some(names) = capture_names.as_ref() {
+                    names[i].clone()
```

Comment:
You can avoid cloning here.. e.g. by using `into_iter()` on the result of `closure_saved_names_of_captured_variables`, then consuming the iterator each time through the map. I don't know of any easy way of using zip here.

Reply:
Addressed in 0cb6f07ef2690b9ad5941c33a0928bf72788829d.